controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -252,7 +252,8 @@ protected void startUp() {
                 clientConfigBuilder.enableTlsToSegmentStore(tlsEnabledForSegmentStore.get());
             }
             
-            ClientConfig clientConfig = clientConfigBuilder.build();
+            // Use one connection per Segment Store to save up resources.
+            ClientConfig clientConfig = clientConfigBuilder.maxConnectionsPerSegmentStore(1).build();
```

Comment:
It would be super nice if we could make this configurable.

Reply:
If having more connections is going to waste resources without adding any performance benefit in this case, I don't see the point of adding yet another configuration parameter. I also don't know how many users will use that knob and, for those who use it, if they will know what to do with it. Overall, in this specific case, I think that 1 is the right setting as changing it can give just resource consumption (leading to OOM) and there is no observed benefit from increasing it.